docs/roadmap.rst

```diff
@@ -0,0 +1,112 @@
+* Simulate Supernovae in the Nebular Phase (Summer 2021)
+ - Currently TARDIS assumes an inner boundary approximation to the
+   photosphere.  New physics is required to handle the breakdown of the
+   photosphere at late times. This involves a detailed treatment of the
+   decay products of the radioactive isotopes in SN ejecta, as well as
+   their deposition in various processes such as Compton scattering,
+   photoabsorption and pair production.
+ - Deposited energy needs to be thermalized by solving the Spencer-Fano
+   equations, resulting in fractions of the energy going into heating,
+   non-thermal excitation and non-thermal ionization.
+ - At late times when the densities are low collisions become too
+   infrequent to quickly de-excite metastable energy levels. Forbidden
+   lines arising from these levels need to be included. Transitions between
+   levels have to include ion-electron collisions.
+
+Website and Documentation
+#########################
+
+* Notebookify (Fall 2021)
+ - TARDIS currently has detailed code and physics documentation
+   (https://tardis-sn.github.io/tardis/)
+ - In order to help new members join the TARDIS development team, we want
+   to create interactive jupyter notebooks that are incorporated into the
+   documentation to better illustrate how the code is used and show how
+   the physics of TARDIS works.
+ - Apply to Google Summer of Docs to get a dedicated, professional
+   documentation writer.
+
+Analysis
+########
+
+* Visualization (Fall 2020)
+ - TARDIS is used to explore potential models for observed supernovae,
+   and it is often difficult to find optimal parameter values.
+ - We are working on an interactive Graphical User Interface for
+   TARDIS that will allow users to visually inspect their models and
+   obtain physical information from the TARDIS simulation easily.
+ - Most of the analysis will be used via Jupyter Notebooks.
+ - This is a Google Summer of Code project
+   (see https://github.com/tardis-sn/tardis/projects/5 ) with time to
+   completion aimed Fall 2020.
+
+Maintenance
+###########
+
+* TARDIS Code Restructure (Fall 2020)
+ - TARDIS needs to be more modularized to facilitate the addition of
+   new physics modules (see https://github.com/tardis-sn/tardis/projects/11 )
+   and improve maintainability.
+ - Google Summer of Code project is currently underway to restructure the
+   model classes that TARDIS uses.
+ - Cython and C code for monte carlo simulation is currently being replaced
+   with Numba to facilitate modularity and debugging.
+
+Performance
+###########
+
+* Numba (Spring 2021)
+ - In the past, TARDIS has used Cython and C for performance-critical parts
+   of the code. However, this requires significant code overhead and is difficult
+   to maintain given that many of our developers are junior researchers who may
+   be less familiar with C code.
+ - As one of TARDIS’ key focuses is its use as an educational tool, we would like
```

Comment:
Part of TARDIS’ mission is to be an educational tool for researchers and developers at all levels. This project aims to minimize the complexity of the code so that it can be understood by a wider audience.